### PR TITLE
Drop iOS 12 support

### PR DESCRIPTION
**Pull request by martinmitrevski**
### 🔗 Issue Links

Resolves https://stream-io.atlassian.net/browse/PBE-4993.

### 🎯 Goal

Drop iOS 12 support, since its usage is pretty low and it makes it harder to maintain it. It will also facilitate the adoption of Swift 6.

### 📝 Summary

Removed starscream, and some iOS < 13 code.

### 🛠 Implementation

Mostly removing code.

### 🎨 Showcase

_Add relevant screenshots and/or videos/gifs to easily see what this PR changes, if applicable._

| Before | After |
| ------ | ----- |
|  img   |  img  |

### 🧪 Manual Testing Notes

_Explain how this change can be tested manually, if applicable._

### ☑️ Contributor Checklist

- [ ] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] This change should be manually QAed
- [ ] Changelog is updated with client-facing changes
- [ ] Changelog is updated with new localization keys
- [ ] New code is covered by unit tests
- [ ] Comparison screenshots added for visual changes
- [ ] Affected documentation updated (docusaurus, tutorial, CMS)

### 🎁 Meme

Alt Text

### CODE REVIEW

**Comment by laevandus** [comment]
There are some comments like "Any because CheckedContinuation<Void, Error> requires iOS 13" what we can fix and clean up as well.

**Comment by martinmitrevski** [comment]
> There are some comments like "Any because CheckedContinuation<Void, Error> requires iOS 13" what we can fix and clean up as well.

Good point, fixed those as well.

**Review by laevandus — APPROVED** [review]

**Comment by nuno-vieira** [comment]
@martinmitrevski We still have a couple of places where we say "Remove this when dropping iOS 12 support" can we tackle those as well?

File: Package.swift [hunk 1]
```diff
@@ -7,7 +7,7 @@ let package = Package(
     name: "StreamChat",
     defaultLocalization: "en",
     platforms: [
-        .iOS(.v12), .macOS(.v10_15)
+        .iOS(.v13), .macOS(.v11)
```
Comment (nuno-vieira) [inline Package.swift:10]: - We need to update the README.me iOS bagde to iOS 13+ (it is still on 11 actually)
- We also need to update the Documentation.docc which we say iOS 12+

Comment (martinmitrevski) [reply Package.swift:10]: done

**Comment by martinmitrevski** [comment]
> @martinmitrevski We still have a couple of places where we say "Remove this when dropping iOS 12 support" can we tackle those as well?
> 
> ChannelMemberListQueryDTO.Line:61, we can remove the additional nil check

Good point, removed those as well.

**Review by nuno-vieira — APPROVED** [review]

**Pull request merged**